### ISSUE TITLE
Over-Simplification of specular pdf in Chapter 12 Rest

### ISSUE THREAD

**Issue by trevordblack** [RayTracing/raytracing.github.io#287]
```
The catch with that approach is that we
need to be careful when we ask for the PDF value and be aware of whether for this evaluation of
`ray_color()` it is diffuse or specular. Fortunately, we know that we should only call the
`pdf_value()` if it is diffuse so we can handle that implicitly.
```

While perfectly specular reflections will not have a pdf, and will exhibit perfect dirac delta reflection. Specular objects will often still have a roughness parameter that smears out the pdf. This is typically modelled with the micro-geometry assumption.

I understand that this is a throw-away line, but this simplification feels over-simple and inaccurate.

Thoughts?

**Comment by trevordblack** [comment]
it also looks like the `is_specular` variable in `scatter_record` is really more of a `dont_use_pdf` variable

**Comment by vchizhov** [comment]
It is inaccurate. The moment you sample randomly due to your fuzziness factor, your distribution is not a Dirac delta anymore, but I guess that was the easiest way to get something "nice" without going into details.

**Comment by trevordblack** [comment]
Resolved in #840
